frontend/src/framework/store/middlewares/error-middleware.ts

```diff
@@ -1,14 +1,27 @@
 import { type Middleware, isRejected } from '@reduxjs/toolkit';
 
+import { actions as authActions } from '~/bundles/auth/store/auth.js';
+import { UserValidationMessage } from '~/bundles/users/users.js';
 import { notificationManager } from '~/framework/notification/notification.js';
 
-const errorMiddleware: Middleware = () => {
+import { type store } from '../store.js';
+
+const errorMiddleware: Middleware = ({
+    dispatch,
+}: {
+    dispatch: typeof store.instance.dispatch;
+}) => {
     return (next) => {
-        return (action) => {
+        return async (action) => {
             if (isRejected(action)) {
                 const { message = 'Something went wrong' } = action.error;
 
-                notificationManager.error(message);
+                if (message === UserValidationMessage.TOKEN_INVALID) {
```

Comment:
I suggest either check status code or add separate error type on the server for authorization errors